src/devices.js

```diff
@@ -234,12 +286,10 @@ function installStep(step) {
                     "Adb is scanning for devices"
                   );
                   function adbWait() {
-                    return adb
-                      .waitForDevice()
-                      .then(resolve)
-                      .catch(() =>
-                        mainEvent.emit("user:connection-lost", adbWait)
-                      );
+                    return adb.hasAccess().then(access => {
```

Comment:
> Also hasAccess returns immediately while waitForDevice waits for the device, is that what you want?

Yes. I found that it feels more intuitive like this. You're shown a popup with a try again button, if you click that, the installer checks once and only continues once you reconnect. Before, if you did not have connection, you would not know what was going on for a couple seconds while waitForDevice timed out.

> Also this will break, as hasAccess uses catch() if the dervice is not found/access see https://github.com/ubports/promise-android-tools/blob/master/src/adb.js#L352 (imo this is wrong way of doing this, it should use then() with a bool variable, catch should only be used if its an error going on)

Read the code you linked to again. It resolves `true` if there's access and `false` if not. It only rejects if there's an error.